Make const-able functions const

Many methods on `InputEvent`, `TimeVal`, `AbsInfo`, could be made const. This would allow code such as

```rust
const SYN_REPORT_EVENT: InputEvent = InputEvent::new(
    &TimeVal::new(0, 0),
    &EventCode::EV_SYN(EV_SYN::SYN_REPORT),
    0,
);
```
Right now, the above code fails with `error[E0015]: calls in statics are limited to constant functions, tuple structs and tuple variants`. `ReadFlags` already has many const methods since it is declared using bitflags, which generates const methods.

--- a/src/enums.rs
+++ b/src/enums.rs
@@ -23,7 +23,7 @@ pub enum EventType {
     EV_MAX = 31,
 }
 
-pub fn int_to_event_type(code: u32) -> Option<EventType> {
+pub const fn int_to_event_type(code: u32) -> Option<EventType> {
     match code {
         0 => Some(EventType::EV_SYN),
         1 => Some(EventType::EV_KEY),
@@ -105,7 +105,7 @@ pub enum EV_REL {
     REL_MAX = 15,
 }
 
-pub fn int_to_ev_rel(code: u32) -> Option<EV_REL> {
+pub const fn int_to_ev_rel(code: u32) -> Option<EV_REL> {
     match code {
         0 => Some(EV_REL::REL_X),
         1 => Some(EV_REL::REL_Y),
@@ -197,7 +197,7 @@ pub enum EV_ABS {
     ABS_MAX = 63,
 }
 
-pub fn int_to_ev_abs(code: u32) -> Option<EV_ABS> {
+pub const fn int_to_ev_abs(code: u32) -> Option<EV_ABS> {
     match code {
         0 => Some(EV_ABS::ABS_X),
         1 => Some(EV_ABS::ABS_Y),
@@ -898,7 +898,7 @@ pub enum EV_KEY {
     BTN_TRIGGER_HAPPY40 = 743,
 }
 
-pub fn int_to_ev_key(code: u32) -> Option<EV_KEY> {
+pub const fn int_to_ev_key(code: u32) -> Option<EV_KEY> {
     match code {
         0 => Some(EV_KEY::KEY_RESERVED),
         1 => Some(EV_KEY::KEY_ESC),
@@ -2119,7 +2119,7 @@ pub enum EV_LED {
     LED_MAX = 15,
 }
 
-pub fn int_to_ev_led(code: u32) -> Option<EV_LED> {
+pub const fn int_to_ev_led(code: u32) -> Option<EV_LED> {
     match code {
         0 => Some(EV_LED::LED_NUML),
         1 => Some(EV_LED::LED_CAPSL),
@@ -2168,7 +2168,7 @@ pub enum EV_SND {
     SND_MAX = 7,
 }
 
-pub fn int_to_ev_snd(code: u32) -> Option<EV_SND> {
+pub const fn int_to_ev_snd(code: u32) -> Option<EV_SND> {
     match code {
         0 => Some(EV_SND::SND_CLICK),
         1 => Some(EV_SND::SND_BELL),
@@ -2204,7 +2204,7 @@ pub enum EV_MSC {
     MSC_MAX = 7,
 }
 
-pub fn int_to_ev_msc(code: u32) -> Option<EV_MSC> {
+pub const fn int_to_ev_msc(code: u32) -> Option<EV_MSC> {
     match code {
         0 => Some(EV_MSC::MSC_SERIAL),
         1 => Some(EV_MSC::MSC_PULSELED),
@@ -2260,7 +2260,7 @@ impl EV_SW {
     pub const SW_MAX: EV_SW = EV_SW::SW_MACHINE_COVER;
 }
 
-pub fn int_to_ev_sw(code: u32) -> Option<EV_SW> {
+pub const fn int_to_ev_sw(code: u32) -> Option<EV_SW> {
     match code {
         0 => Some(EV_SW::SW_LID),
         1 => Some(EV_SW::SW_TABLET_MODE),
@@ -2320,7 +2320,7 @@ pub enum EV_SYN {
     SYN_MAX = 15,
 }
 
-pub fn int_to_ev_syn(code: u32) -> Option<EV_SYN> {
+pub const fn int_to_ev_syn(code: u32) -> Option<EV_SYN> {
     match code {
         0 => Some(EV_SYN::SYN_REPORT),
         1 => Some(EV_SYN::SYN_CONFIG),
@@ -2357,7 +2357,7 @@ impl EV_REP {
     pub const REP_MAX: EV_REP = EV_REP::REP_PERIOD;
 }
 
-pub fn int_to_ev_rep(code: u32) -> Option<EV_REP> {
+pub const fn int_to_ev_rep(code: u32) -> Option<EV_REP> {
     match code {
         0 => Some(EV_REP::REP_DELAY),
         1 => Some(EV_REP::REP_PERIOD),
@@ -2390,7 +2390,7 @@ pub enum InputProp {
     INPUT_PROP_MAX = 31,
 }
 
-pub fn int_to_input_prop(code: u32) -> Option<InputProp> {
+pub const fn int_to_input_prop(code: u32) -> Option<InputProp> {
     match code {
         0 => Some(InputProp::INPUT_PROP_POINTER),
         1 => Some(InputProp::INPUT_PROP_DIRECT),
@@ -2450,7 +2450,7 @@ impl EV_FF {
     pub const FF_STATUS_MAX: EV_FF = EV_FF::FF_STATUS_PLAYING;
 }
 
-pub fn int_to_ev_ff(code: u32) -> Option<EV_FF> {
+pub const fn int_to_ev_ff(code: u32) -> Option<EV_FF> {
     match code {
         0 => Some(EV_FF::FF_STATUS_STOPPED),
         1 => Some(EV_FF::FF_STATUS_PLAYING),
@@ -2531,7 +2531,7 @@ pub enum BusType {
     BUS_INTEL_ISHTP = 31,
 }
 
-pub fn int_to_bus_type(code: u32) -> Option<BusType> {
+pub const fn int_to_bus_type(code: u32) -> Option<BusType> {
     match code {
         1 => Some(BusType::BUS_PCI),
         2 => Some(BusType::BUS_ISAPNP),

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -151,7 +151,7 @@ pub struct AbsInfo {
 }
 
 impl AbsInfo {
-    pub fn from_raw(absinfo: libc::input_absinfo) -> AbsInfo {
+    pub const fn from_raw(absinfo: libc::input_absinfo) -> AbsInfo {
         AbsInfo {
             value: absinfo.value,
             minimum: absinfo.minimum,
@@ -162,7 +162,7 @@ impl AbsInfo {
         }
     }
 
-    pub fn as_raw(&self) -> libc::input_absinfo {
+    pub const fn as_raw(&self) -> libc::input_absinfo {
         libc::input_absinfo {
             value: self.value,
             minimum: self.minimum,
@@ -205,7 +205,7 @@ impl TryInto<SystemTime> for TimeVal {
 }
 
 impl TimeVal {
-    pub fn new(tv_sec: time_t, tv_usec: suseconds_t) -> TimeVal {
+    pub const fn new(tv_sec: time_t, tv_usec: suseconds_t) -> TimeVal {
         const MICROS_PER_SEC: suseconds_t = 1_000_000;
         TimeVal {
             tv_sec: tv_sec + tv_usec / MICROS_PER_SEC,
@@ -213,14 +213,14 @@ impl TimeVal {
         }
     }
 
-    pub fn from_raw(timeval: &libc::timeval) -> TimeVal {
+    pub const fn from_raw(timeval: &libc::timeval) -> TimeVal {
         TimeVal {
             tv_sec: timeval.tv_sec,
             tv_usec: timeval.tv_usec,
         }
     }
 
-    pub fn as_raw(&self) -> libc::timeval {
+    pub const fn as_raw(&self) -> libc::timeval {
         libc::timeval {
             tv_sec: self.tv_sec,
             tv_usec: self.tv_usec,
@@ -239,7 +239,7 @@ pub struct InputEvent {
 }
 
 impl InputEvent {
-    pub fn new(timeval: &TimeVal, code: &EventCode, value: i32) -> InputEvent {
+    pub const fn new(timeval: &TimeVal, code: &EventCode, value: i32) -> InputEvent {
         InputEvent {
             time: *timeval,
             event_code: *code,

--- a/tools/make-event-names.py
+++ b/tools/make-event-names.py
@@ -134,7 +134,7 @@ def print_enums_convert_fn(bits, prefix):
     if not hasattr(bits, prefix):
         return
 
-    print("pub fn %s(code: u32) -> Option<%s> {" %
+    print("pub const fn %s(code: u32) -> Option<%s> {" %
           ("int_to_" + convert(fn_name), fn_name))
     print("    match code {")
     for val, names in list(getattr(bits, prefix).items()):

